src/config-loader/config-loader.ts

```diff
@@ -52,294 +52,267 @@ export interface Config {
-			root = path.resolve(process.argv[2]) || __dirname;
-			configPromise = loadConfigFile(path.join(root, 'config.json'));
-		} else if (_.isString(config)) {
-			root = path.dirname(config);
-			configPromise = loadConfigFile(config);
-		} else if (_.isObject(config)) {
-			root = process.cwd();
-			configPromise = config;
-		} else {
-			return Bluebird.reject(
-				new Error(`Invalid type for config '${typeof config}'`),
-			);
-		}
-		const resolvePath = (s: string): string => {
-			if (path.isAbsolute(s)) {
-				return s;
-			}
-			return path.join(root, s);
-		};
+				console.info('Loading application config');
+				let root: string;
+				let configObj: Config;
+				if (config == null) {
+					root = path.resolve(process.argv[2]) || __dirname;
+					configObj = await loadConfigFile(path.join(root, 'config.json'));
+				} else if (_.isString(config)) {
+					root = path.dirname(config);
+					configObj = await loadConfigFile(config);
+				} else if (_.isObject(config)) {
+					root = process.cwd();
+					configObj = config;
+				} else {
+					throw new Error(`Invalid type for config '${typeof config}'`);
+				}
+				const resolvePath = (s: string): string => {
+					if (path.isAbsolute(s)) {
+						return s;
+					}
+					return path.join(root, s);
+				};
 
-		return Bluebird.resolve(configPromise)
-			.then(configObj =>
-				Bluebird.map(configObj.models, model =>
-					Bluebird.try<string | undefined>(() => {
-						if (model.modelFile != null) {
-							return fs.promises.readFile(resolvePath(model.modelFile), 'utf8');
-						}
-						return model.modelText;
-					})
-						.then(modelText => {
-							model.modelText = modelText;
-							if (_.isString(model.customServerCode)) {
-								model.customServerCode = resolvePath(model.customServerCode);
-							}
-						})
-						.then(() => {
-							if (model.migrations == null) {
-								model.migrations = {};
-							}
-							const migrations = model.migrations;
+				await Bluebird.map(configObj.models, async model => {
+					if (model.modelFile != null) {
+						model.modelText = await fs.promises.readFile(
+							resolvePath(model.modelFile),
+							'utf8',
+						);
+					}
+					if (_.isString(model.customServerCode)) {
+						model.customServerCode = resolvePath(model.customServerCode);
+					}
+					if (model.migrations == null) {
+						model.migrations = {};
+					}
+					const migrations = model.migrations;
 
-							if (model.migrationsPath) {
-								const migrationsPath = resolvePath(model.migrationsPath);
-								delete model.migrationsPath;
+					if (model.migrationsPath) {
+						const migrationsPath = resolvePath(model.migrationsPath);
+						delete model.migrationsPath;
 
-								return Bluebird.resolve(fs.promises.readdir(migrationsPath))
-									.map(filename => {
-										const filePath = path.join(migrationsPath, filename);
-										const [migrationKey] = filename.split('-', 1);
+						await Bluebird.resolve(fs.promises.readdir(migrationsPath)).map(
```

Comment:
How about
```suggestion
						await Bluebird.map(await fs.promises.readdir(migrationsPath), async filename => {
```
similarly to what you did a bit earlier?